### PR TITLE
Fix object property completion when have hyphen

### LINKED ISSUE #1804: Object property with quotes will not show in auto complete hints
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

## Info

- Platform: macOS
- Vetur version: 0.24.0
- VS Code version: 1.43.1

## Problem

Object property with quotes will not show in auto complete hints

WX20200324-173222

## Reproducible Case

create a `.vue` file and a `.js` file
a.vue
```html
<template>
</template>
<script>
import a from './a';
a.
</script>
```

a.js
```
export default {
  中文: 1,
  pp: 1,
  'pp-1': 1,
  '中文-1': 1
};
```

### PR DESCRIPTION
**Pull request by yoyo930021**
Fixed #1804

### CODE REVIEW

**Comment by littlee** [comment]
快点合并

**Comment by octref** [comment]
I have a simpler fix. The ranking is not preserved but that's a VS Code issue: https://github.com/microsoft/vscode/issues/103682

File: server/src/modes/script/javascript.ts [hunk 1]
```diff
@@ -157,17 +158,29 @@ export async function getJavascriptMode(
...
+        items: entries.map(entry => {
           const range = entry.replacementSpan && convertRange(scriptDoc, entry.replacementSpan);
+          const filterText = entry.insertText && range && entry.insertText[0] === '[' ? '.' + entry.name : undefined;
           const { label, detail } = calculateLabelAndDetailTextForPathImport(entry);
+          const kind = toCompletionItemKind(entry.kind);
+          const insertTextFormat =
+            kind === CompletionItemKind.Function || kind === CompletionItemKind.Method
+              ? InsertTextFormat.Snippet
+              : undefined;
+          const insertText = entry.insertText || entry.name;
```
Comment (octref) [inline server/src/modes/script/javascript.ts:170]: No need to do this. If `entry.insertText` is undefined just pass it in.

**Comment by yoyo930021** [comment]
@octref I forget why write this code.
I just remember I reference [this issue](https://github.com/microsoft/vscode/blob/63d8f6177289271fcaa375d3c85193dc390113b5/extensions/typescript-language-features/src/languageFeatures/completions.ts#L175).

I found that it has changed a lot.

**Comment by octref** [comment]
I think it's ok to merge now. The `typescript-language-features` extension will always be changing. No worries.

**Pull request merged**